src/gui/ChangeMasterKeyWidget.cpp

```diff
@@ -130,6 +164,23 @@ void ChangeMasterKeyWidget::generateKey()
         m_key.addKey(fileKey);
     }
 
+#ifdef WITH_XC_YUBIKEY
+    if (m_ui->challengeResponseGroup->isChecked()) {
+        int i = m_ui->comboChallengeResponse->currentIndex();
+        i = m_ui->comboChallengeResponse->itemData(i).toInt();
+
+        if (0 == i) {
+            m_ui->messageWidget->showMessage(tr("Changing master key failed: no YubiKey inserted."),
+                                             MessageWidget::Error);
+            return;
+        }
+        bool blocking = i & true;
+        int slot      = i >> 1;
+        auto key      = QSharedPointer<YkChallengeResponseKey>(new YkChallengeResponseKey(slot, blocking));
```

Comment:
I'll add a comment. It's basically just a dirty hack to encode more than one piece of information in a combo box entry.